brave/brave-ios
# Fix #3939: Add Screen Time support.

## Summary of Changes

Old PR by our contributor here https://github.com/brave/brave-ios/pull/7639


This pull request fixes #3939 
Security review https://github.com/brave/reviews/issues/1383

## Submitter Checklist:


## Test Plan:

1. Verify that when you stay on a website for a bit it is tracked in Screen Time section in iOS settings.
2. Verify that after setting a time limit for a website, after you reach the limit you are not able to access it anymore.
3. Verify that websites are not tracked in private browsing mode

Disabled tests:
1. Turn off the feature - verify that websites are not tracked again
2. Turn off the feature - verify that the website where you reached the time limit can be accessed

## Screenshots:

All the in-website UI is Apple controlled.
<img width="516" alt="Zrzut ekranu 2023-08-29 o 13 40 11" src="https://github.com/brave/brave-ios/assets/6950387/bc30adb6-159a-4f76-a8b7-2a8040fa40b3">

## Reviewer Checklist:

  - `QA/(Yes|No)`
  - `bug` / `enhancement`
